docs/en/writing-migrations.rst

```diff
@@ -1159,6 +1160,232 @@ involved::
+                  // Name will be auto-generated like 'inventory_chk_a1b2c3d4'
+                  ->addCheckConstraint(
+                      $this->checkConstraint()
+                          ->setExpression('quantity >= 0')
+                  )
+                  ->save();
+        }
+    }
+
+Complex Check Constraints
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Check constraints can reference multiple columns and use complex SQL expressions::
+
+    <?php
+
+    use Migrations\BaseMigration;
+
+    class MyNewMigration extends BaseMigration
+    {
+        /**
+         * Migrate Up.
+         */
+        public function up(): void
+        {
+            $table = $this->table('date_ranges');
+            $table->addColumn('start_date', 'date')
+                  ->addColumn('end_date', 'date')
+                  ->addColumn('discount', 'decimal', ['precision' => 5, 'scale' => 2])
+                  ->addCheckConstraint('valid_date_range', 'end_date >= start_date')
+                  ->addCheckConstraint('valid_discount', 'discount BETWEEN 0 AND 100')
+                  ->save();
+        }
+    }
+
+Checking if a Check Constraint Exists
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can verify if a check constraint exists using the ``hasCheckConstraint()`` method::
+
+    <?php
+
+    use Migrations\BaseMigration;
+
+    class MyNewMigration extends BaseMigration
+    {
+        /**
+         * Migrate Up.
+         */
+        public function up(): void
+        {
+            $table = $this->table('products');
+            $exists = $table->hasCheckConstraint('price_positive');
+            if ($exists) {
+                // do something
+            } else {
+                $table->addCheckConstraint('price_positive', 'price > 0')
+                      ->save();
+            }
+        }
+    }
+
+Dropping a Check Constraint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To remove a check constraint, use the ``dropCheckConstraint()`` method with the
+constraint name::
+
+    <?php
+
+    use Migrations\BaseMigration;
+
+    class MyNewMigration extends BaseMigration
+    {
+        /**
+         * Migrate Up.
+         */
+        public function up(): void
+        {
+            $table = $this->table('products');
+            $table->dropCheckConstraint('price_positive')
+                  ->save();
+        }
+
+        /**
+         * Migrate Down.
+         */
+        public function down(): void
+        {
+            $table = $this->table('products');
+            $table->addCheckConstraint('price_positive', 'price > 0')
+                  ->save();
+        }
+    }
+
+.. note::
+
+    Like other table operations, ``dropCheckConstraint()`` requires ``save()``
+    to be called to execute the change.
+
+Database-Specific Behavior
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**MySQL (8.0.16+)**
+
+Check constraints are fully supported. MySQL stores constraint metadata in the
+``INFORMATION_SCHEMA.CHECK_CONSTRAINTS`` table.
```

Comment:
Migration diffs will probably drop check constraints, as we don't currently have any reflection support for check constraints in cakephp/database